app/workers/internal_worker.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class InternalWorker
```

Comment:
Should we rename this to `IntervalWorker`? Basically we'll want two behaviours: one for scheduled receivers (runs once), and one for interval receivers (reschedules itself at the end).